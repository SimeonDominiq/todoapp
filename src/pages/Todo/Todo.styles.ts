import styled from "styled-components"

export const TodoWrapper = styled.div`
  height: 100vh;
  width: 100%;
  background: #e8e8e8;
  display: flex;
  align-items: center;
  justify-content: center;
`

export const TodoContainer = styled.div`
  width: 25%;
  background: #fff;
  padding: 25px 20px;
`
