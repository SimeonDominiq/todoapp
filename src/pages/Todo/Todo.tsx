import { useSelector } from "react-redux"
import { RootState } from "../../store/store"
import TodoListGroup from "src/components/TodoListGroup/TodoListGroup"
import { ITodoGroup } from "src/types"
import RandomFact from "src/components/RandomFact/RandomFact"

import { TodoContainer, TodoWrapper } from "./Todo.styles"

const Todo = () => {
  const todoGroups = useSelector((state: RootState) => state.todoGroups)
  const checkGroupIsCompleted = (group: ITodoGroup) =>
    group.todoItems.every((item) => item.isCompleted === true)
  const allPhaseCompleted = todoGroups.every(checkGroupIsCompleted)

  return (
    <TodoWrapper>
      <TodoContainer>
        {allPhaseCompleted ? (
          <RandomFact />
        ) : (
          <>
            <h2>My starup progress</h2>
            {todoGroups?.map((todoGroup) => (
              <TodoListGroup
                key={todoGroup.groupId}
                {...{
                  todoGroup,
                }}
              />
            ))}
          </>
        )}
      </TodoContainer>
    </TodoWrapper>
  )
}

export default Todo
