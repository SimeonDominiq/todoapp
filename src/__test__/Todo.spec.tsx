import Todo from "../pages/Todo/Todo"

describe("App: Components: Todo Layout", () => {
  it("matches snapshot", () => {
    expect(Todo).toMatchSnapshot()
  })
})
