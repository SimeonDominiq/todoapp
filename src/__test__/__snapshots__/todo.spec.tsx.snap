// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Components: Todo Layout matches snapshot 1`] = `[Function]`;
